fhttp/http_client.go

```diff
@@ -430,7 +431,7 @@ func (c *FastClient) Close() int {
 // NewFastClient makes a basic, efficient http 1.0/1.1 client.
 // This function itself doesn't need to be super efficient as it is created at
 // the beginning and then reused many times.
-func NewFastClient(o *HTTPOptions) Fetcher {
+func NewFastClient(o *HTTPOptions) *FastClient {
```

Comment:
The idea there was to hide implementation details, that you get just a Fetcher either way but maybe this is ok - depending on the answer to my other question (why do you need the details)

Reply:
In the `TestPayloadForFastClient` test, I'm looking for the `client.req` field to check If the body is applied on FastClient as I expected. So I needed to receive `*FastClient` typed object to look 'req' field.

Reply:
you could just check it is echoed back?

this being said we can make that signature change as the otje std cloent already retuens the concrete type

Reply:
Okay, I take `Fetcher` back. But at `TestPayloadForFastClient` test, my point is to see do I send body to server as I expected. I mean that I'm checking the headers like `content type`,`content length` and `method`. If I use echo back for `TestPayloadForFastClient` I won't able to cover the method type.

Reply:
Btw if you are saying that instead of `TestPayloadForClient` and `TestPayloadForFastClient` tests, we should use e2e, its okay for me as well. I just want to know that is why I was not trying to use echo back with above comment.

Reply:
Any comment for here :)